mod.ts

```diff
@@ -0,0 +1,81 @@
+const canElideFrames = 'captureStackTrace' in Error
+const startStackFrames = new WeakMap()
+
+interface Result {
+  name: 'AssertionError' | 'AssertionResult'
+  ok: boolean
+  toJSON(...args: unknown[]): Record<string, unknown>
+}
+
+export class AssertionError<T> extends Error implements Result {
+  [key: string]: unknown
```

Comment:
Is it possible to move this into the `Result` interface?

Reply:
Moving this to the interface causes type errors because then the class doesn't implement this type.